logger/Logger.cxx

```diff
@@ -144,16 +145,15 @@ map<Verbosity, VSpec> Logger::fVerbosities =
     { Verbosity::user4,    VSpec::Make(VSpec::Info::severity)                                                                                        }
 };
 
-Logger::Logger(Severity severity, Verbosity verbosity, const string& file, const string& line, const string& func)
+Logger::Logger(Severity severity, Verbosity verbosity, std::string_view file, char const* line, char const* func)
     : fTimeCalculated(false)
 {
     if (!fIsDestructed) {
         size_t pos = file.rfind("/");
-
         // fInfos.timestamp is filled conditionally
         // fInfos.us is filled conditionally
-        fInfos.process_name = fProcessName;
-        fInfos.file = file.substr(pos + 1);
+        fInfos.process_name = fProcessName.c_str();
+        fInfos.file = file.substr(pos + 1).data();
```

Comment:
I guess converting `string_view::data()` to a null-terminated "C-String" is an error in any case. (If you look at the diff, one can see the relevant hint, `data() != c_str()` in terms of null-termination guarantees) Better to keep the `char const * file` argument then as it - if the reader knows about the C-String-World - documents clearly that we expect a null-terminated string. By accepting a `string_view` the interface does not express this assumption/requirement any more.

Reply:
Fair point. For now I simply use a string view everywhere.